src/baseClasses/ApplicationBase.ts

```diff
@@ -758,5 +763,33 @@ export default class ApplicationBase {
       (location.hostname === "localhost" ||
         location.hostname === "127.0.0.1");
   }
+  private _calculateLocale(portal) {
```

Comment:
I've got some logic in Slider right now that does this manually: https://devtopia.esri.com/WebGIS/arcgis-portal-app-templates/blob/163ecb06a5801df971ef7f784fd216d3f811c237/instant/slider/src/init.ts#L136

I have two questions:
1. Do you think I should remove that logic from Slider when this goes in?
2. Should we include the browser language in the order of precedence? (This is what I'm doing in the landing page: https://devtopia.esri.com/WebGIS/arcgis-template-landing/blob/0e21fcdd102f8eb811a04def1e3d1bcecb526a82/src/app/utils/t9nUtils.ts#L14)

Reply:
@Csmith246
1. Yes I think once we get it set to match the document below we'll want all the apps to use the same logic so that if we have to make changes in the future we can do it quickly and in one place. 
2.  i think the jsapi setLocale will fallback to the browser locale. I have the logic in this PR In basic and i'm going to test it today. It needs to match the order Blake outlined in this document: 
https://confluencewikidev.esri.com/display/AGO/Locale+Loading

I'll test in Basic today and may have changes to this PR